### ISSUE TITLE
LDAP user filter not used after mapping

### ISSUE THREAD

**Issue by etemam** [owncloud/core#20975]
Hi,

we have a little problem with owncloud. We bind it to an openLDAP server. We have a field x-ServicesEnable that must have owncloud as values to enable a user connecting to owncloud instance. Therefore we put the following LDAP configuration parameters:
```
ldapUserFilter = (&(x-ServicesEnable=owncloud)(objectClass=inetOrgPerson))
ldapLoginFilter = (&(x-ServicesEnable=owncloud)(uid=%uid))
```
and everything is working well.

Except that if we deactivate the service for a user, owncloud still find him:

```
[www@oc ~]# sudo -u www php occ ldap:check-user DEACTIVATE_USER
The user is still available on LDAP.
```

### Step to reproduce:

  1. Connect owncloud to openLDAP
  2. Add both parameters:
```
ldapUserFilter = (&(AFIELD=AVALUE)(objectClass=inetOrgPerson))
ldapLoginFilter = (&(AFIELD=AVALUE)(uid=%uid))
```
 3. Create a user with AFIELD parameter's value equal to AVALUE
 4. Open a session on owncloud with this user
 5. Delete AFIELD parameter for this user
 6. User can still connect....

### Configuration

  *. Owncloud 8.2
  *. Postgresql 9.4
  *. Apache 2.4
  *. Freebsd 9.3
  *. Redis as cache
  *. PHP 5.6

Thanks for help,

PS: we achieve to solve the problem by creating a replica that replicate only owncloud available users. But we cannot maintain this solution for a long time

**Comment by PVince81** [comment]
@blizzz

**Comment by blizzz** [comment]
@etemam could you check whether https://github.com/owncloud/core/pull/18212 fixes your issue?

**Comment by blizzz** [comment]
I assume it's fixed with #18212, please shout if it is not the case.
